patches/tModLoader/Terraria/UI/ItemSlot.cs.patch

```diff
@@ -44,14 +44,61 @@
  						}
  
  						Recipe.FindRecipes();
-@@ -712,20 +_,24 @@
- 			if (inv[slot].type <= 0)
- 				return;
+@@ -683,28 +_,34 @@
+ 		}
  
+ 		private static bool LeftClick_SellOrTrash(Item[] inv, int context, int slot) {
+-			bool flag = false;
+-			bool result = false;
+-			if (NotUsingGamepad && Options.DisableLeftShiftTrashCan) {
+-				if ((uint)context <= 4u || context == 7)
+-					flag = true;
+-
+-				if (ControlInUse && flag) {
+-					SellOrTrash(inv, context, slot);
+-					result = true;
+-				}
++			Player player = Main.player[Main.myPlayer];
 +			if (PlayerHooks.ShiftClickSlot(player, inv, context, slot)) {
-+			}
```

Comment:
This is an unnecessary large patch. Avoid moving and deleting vanilla code, and, especially, indenting it. Here you could've just used a short-circuit return.